pkg/cri/server/restart.go

```diff
@@ -441,9 +441,8 @@ func (c *criService) loadImages(ctx context.Context, cImages []containerd.Image)
 		unpacked, err := i.IsUnpacked(ctx, snapshotter)
 		if err != nil {
 			log.G(ctx).WithError(err).Warnf("Failed to check whether image is unpacked for image %s", i.Name())
-			continue
 		}
-		if !unpacked {
+		if err == nil && !unpacked {
```

Comment:
I'm worried that ignoring the error leads to using invalid snapshots.
Why not start containerd after ensuring the readiness of the snapshotter plugin?

Reply:
> I'm worried that ignoring the error leads to using invalid snapshots. Why not start containerd after ensuring the readiness of the snapshotter plugin?

In this scenario,  `ctr images list` can list images,  so I don't think it should have any effect on getting the images list.

Reply:
I agree with @ktock. Ignoring any errors (not just specific errors) from the CRI plugin itself (not in frontend CLIs) would affect other cases.

You should start your snapshotter before containerd. Wouldn't it be possible?